src/app/forms/custom-sections-v2/adr/adr.component.ts

```diff
@@ -137,10 +138,36 @@ export class AdrComponent extends EditBaseComponent implements OnInit, OnDestroy
 		techRecord_adrDetails_adrCertificateNotes: this.fb.control<string | null>(null, [
 			this.commonValidators.maxLength(1500, 'ADR Certificate Notes must be less than or equal to 1500 characters'),
 		]),
+		// Tank Details > Tank Inspections
+		techRecord_adrDetails_tank_tankDetails_tc2Details_tc2Type: this.fb.control<string | null>('initial'),
+		techRecord_adrDetails_tank_tankDetails_tc2Details_tc2IntermediateApprovalNo: this.fb.control<string | null>(null, [
+			this.adrValidators.requiredWithTankOrBattery('TC2: Certificate number is required with ADR body type'),
+			this.commonValidators.maxLength(70, 'TC2: Certificate Number must be less than or equal to 70 characters'),
+		]),
+		techRecord_adrDetails_tank_tankDetails_tc2Details_tc2IntermediateExpiryDate: this.fb.control<string | null>(null, [
+			this.commonValidators.date('TC2: Expiry date'),
+			this.adrValidators.requiredWithTankOrBattery('TC2: Expiry date is required with ADR body type'),
+		]),
+		techRecord_adrDetails_tank_tankDetails_tc3Details: this.fb.array<FormGroup>([]),
+
+		// Miscellaneous
+		techRecord_adrDetails_memosApply: this.fb.control<string | null>(null),
+		techRecord_adrDetails_m145Statement: this.fb.control<boolean>(false),
+
+		// Battery List
+		techRecord_adrDetails_listStatementApplicable: this.fb.control<string | null>(null, [
+			this.adrValidators.requiredWithBattery('Battery list applicable is required with ADR body type', true),
+		]),
+		techRecord_adrDetails_batteryListNumber: this.fb.control<string | null>(null, [
+			this.adrValidators.requiredWithBatteryListApplicable('Reference number is required with Battery List Applicable'),
```

Comment:
```suggestion
			this.adrValidators.requiredWithBatteryListApplicable('Reference number is required with Battery list applicable'),
```

ACs have correct label for this one